worlds/yoshisisland/Locations.py

```diff
@@ -0,0 +1,352 @@
+from typing import List, Tuple, Optional, Callable, NamedTuple
+from BaseClasses import MultiWorld
+from .level_logic import YoshiLogic
+
+
+class LocationData(NamedTuple):
+    region: str
+    name: str
+    code: Optional[int]
+    LevelID: int
+    rule: Callable = lambda state: True
+
+
+def get_locations(world) -> Tuple[LocationData, ...]:

```

Comment:
```suggestion
def get_locations(world) -> List[LocationData]:
```
The return value is a List, not a Tuple